src/lib/OpenEXR/ImfStandardAttributes.h

```diff
@@ -249,7 +689,28 @@ IMF_STD_ATTRIBUTE_DEF (timeCode, TimeCode, TimeCode)
 
 
 //
-// wrapmodes -- determines how texture map images are extrapolated.
+// timecodeRate -- the timecode rate of the image sequence to which
+// the image belongs, specified in timecodes per second
+//
+
+IMF_STD_ATTRIBUTE_DEF (timecodeRate, TimecodeRate, int)
+
+
+//
+// imageCounter -- an image number
+//
+// For a sequence of images, the image number increases 
+// when the images are accessed in the intended play order. Can be
+// used to uniquely identify frames of high-speed photography that
+// would have identical timecode.
+//
```

Comment:
Can the image number be expected to correspond to a film edge code, if an edge code is known? Or do image numbers are an index in a sequence, and start at zero, or are they nothing more than what the description indicates?

Reply:
There is a separate keycode attribute in then spec, and imageCounter is not tied to it.

imageCounter numbers impose an ordering, which conceivably could be sparse, though in practice I have never seen that. The actual text of 2065-4 has to put that conceivable use (high-speed photography) into an informational note so that the rest of the definition can be unambiguously normative. I think we suggest their use more strongly with our "Can be used..." but we don't force non-sparse. My saved email on the topic isn't particularly informative, but I suspect that we felt we gave people more power by imposing fewer constraints.

![image](https://user-images.githubusercontent.com/1134845/103465878-aa2e3280-4d0d-11eb-8ac2-28e25fa9c402.png)

Does that answer resolve your issue?

Reply:
Yes. Briefly indicating that the numbering can be sparse would be sufficient to alleviate my first-reading confusion. FWIW, this situation comes up in computational vision. For example, high speed capture on stage might drop frames due to packet ordering and real time constraints. I used the number in one system to match frames between a capture server and a processed image server.